filecoin-proofs/src/api/seal.rs

```diff
@@ -0,0 +1,278 @@
+        &compound_public_params.vanilla_params,
+        &replica_id,
+        &mut data,
+        Some(data_tree),
+    )?;
+
+    let comm_r = commitment_from_fr::<Bls12>(tau.comm_r.into());
+
+    info!("seal_pre_commit:end");
+
+    Ok(SealPreCommitOutput {
+        comm_r,
+        comm_d,
+        p_aux,
+        t_aux,
+    })
+}
+
+/// Generates a proof for the pre committed sector.
+#[allow(clippy::too_many_arguments)]
+pub fn seal_commit<T: AsRef<Path>>(
+    porep_config: PoRepConfig,
+    _cache_path: T,
+    prover_id: ProverId,
+    sector_id: SectorId,
+    ticket: Ticket,
+    seed: Ticket,
+    pre_commit: SealPreCommitOutput,
+    piece_infos: &[PieceInfo],
+) -> error::Result<SealCommitOutput> {
+    info!("seal_commit:start");
+
+    let SealPreCommitOutput {
+        comm_d,
+        comm_r,
+        p_aux,
+        t_aux,
+    } = pre_commit;
+    let comm_r_safe = as_safe_commitment(&comm_r, "comm_r")?;
+    let comm_d_safe = <DefaultPieceHasher as Hasher>::Domain::try_from_bytes(&comm_d)?;
+
+    let replica_id = generate_replica_id::<DefaultTreeHasher, _>(
+        &prover_id,
+        sector_id.into(),
+        &ticket,
+        comm_d_safe,
+    );
+
+    let seed_fr = bytes_into_fr::<Bls12>(&seed).map(Into::into)?;
+
+    let public_inputs = stacked::PublicInputs {
+        replica_id,
+        tau: Some(stacked::Tau {
+            comm_d: comm_d_safe,
+            comm_r: comm_r_safe,
+        }),
+        k: None,
+        seed: seed_fr,
+    };
+
+    let private_inputs =
+        stacked::PrivateInputs::<DefaultTreeHasher, DefaultPieceHasher> { p_aux, t_aux };
+
+    let groth_params = get_stacked_params(porep_config)?;
+
+    info!(
+        "got groth params ({}) while sealing",
+        u64::from(PaddedBytesAmount::from(porep_config))
+    );
+
+    let compound_setup_params = compound_proof::SetupParams {
+        vanilla_params: &setup_params(
+            PaddedBytesAmount::from(porep_config),
+            usize::from(PoRepProofPartitions::from(porep_config)),
+        ),
+        engine_params: &(*JJ_PARAMS),
+        partitions: Some(usize::from(PoRepProofPartitions::from(porep_config))),
+    };
+
+    let compound_public_params = StackedCompound::setup(&compound_setup_params)?;
+
+    let proof = StackedCompound::prove(
+        &compound_public_params,
+        &public_inputs,
+        &private_inputs,
+        &groth_params,
+    )?;
+
+    let mut buf = Vec::with_capacity(
+        SINGLE_PARTITION_PROOF_LEN * usize::from(PoRepProofPartitions::from(porep_config)),
+    );
+
+    proof.write(&mut buf)?;
+
+    // Verification is cheap when parameters are cached,
+    // and it is never correct to return a proof which does not verify.
+    verify_seal(
+        porep_config,
+        comm_r,
+        comm_d,
+        prover_id,
+        sector_id,
+        ticket,
+        seed,
+        &buf,
+        piece_infos,
+    )
+    .expect("post-seal verification sanity check failed");
```

Comment:
BLOCKING: This does not actually assert that the proof is valid. You need to check that the value of this expression equals `true`.

Reply:
⭐️

Reply:
fixed